pydicom/tests/test_rle_pixel_data.py

```diff
@@ -1,126 +1,762 @@
+        assert (2155905152, 4294967295, 2155905152) == tuple(arr[35, 50, :])
+        assert (0, 0, 4294967295) == tuple(arr[45, 50, :])
+        assert (2155905152, 2155905152, 4294967295) == tuple(arr[55, 50, :])
+        assert (0, 0, 0) == tuple(arr[65, 50, :])
+        assert (1077952576, 1077952576, 1077952576) == tuple(arr[75, 50, :])
+        assert (3233857728, 3233857728, 3233857728) == tuple(arr[85, 50, :])
+        assert (4294967295, 4294967295, 4294967295) == tuple(arr[95, 50, :])
+
+    def test_pixel_array_32bit_3sample_2f(self):
+        """Test pixel_array for 32-bit, 3, sample/pixel, 2 frame."""
+        ds = dcmread(SC_RLE_32_2F)
+        assert ds.file_meta.TransferSyntaxUID == RLELossless
+        assert ds.BitsAllocated == 32
+        assert ds.SamplesPerPixel == 3
+        assert ds.NumberOfFrames == 2
+        arr = ds.pixel_array
+        ref = _get_pixel_array(SC_EXPL_LITTLE_32_2F)
+
+        assert np.array_equal(ds.pixel_array, ref)
+
+        # Frame 1
+        assert (4294967295, 0, 0) == tuple(arr[0, 5, 50, :])
+        assert (4294967295, 2155905152, 2155905152) == tuple(arr[0, 15, 50, :])
+        assert (0, 4294967295, 0) == tuple(arr[0, 25, 50, :])
+        assert (2155905152, 4294967295, 2155905152) == tuple(arr[0, 35, 50, :])
+        assert (0, 0, 4294967295) == tuple(arr[0, 45, 50, :])
+        assert (2155905152, 2155905152, 4294967295) == tuple(arr[0, 55, 50, :])
+        assert (0, 0, 0) == tuple(arr[0, 65, 50, :])
+        assert (1077952576, 1077952576, 1077952576) == tuple(arr[0, 75, 50, :])
+        assert (3233857728, 3233857728, 3233857728) == tuple(arr[0, 85, 50, :])
+        assert (4294967295, 4294967295, 4294967295) == tuple(arr[0, 95, 50, :])
+
+        # Frame 2 is frame 1 inverted
+        assert np.array_equal((2**ds.BitsAllocated - 1) - arr[1], arr[0])
+
+
+# Tests for rle_handler module with Numpy available
+@pytest.mark.skipif(not HAVE_NP, reason='Numpy is not available')
+class TestNumpy_GetPixelData(object):
+    """Tests for rle_handler.get_pixeldata with numpy."""
+    def test_no_pixel_data_raises(self):
+        """Test get_pixeldata raises if dataset has no PixelData."""
+        ds = dcmread(MR_RLE_1F)
+        del ds.PixelData
+        assert 'PixelData' not in ds
+        # Should probably be AttributeError instead
+        with pytest.raises(TypeError, match='No pixel data found'):
+            get_pixeldata(ds)
+
+    def test_unknown_pixel_representation_raises(self):
+        """Test get_pixeldata raises if unsupported PixelRepresentation."""
+        ds = dcmread(MR_RLE_1F)
+        ds.PixelRepresentation = 2
+        # Should probably be NotImplementedError instead
+        with pytest.raises(TypeError, match="format='bad_pixel_repr"):
+            get_pixeldata(ds)
+
+    def test_unsupported_syntaxes_raises(self):
+        """Test get_pixeldata raises if unsupported Transfer Syntax."""
+        ds = dcmread(MR_EXPL_LITTLE_1F)
+        # Typo in exception message
```

Comment:
I guess you will fix the typo itself in another PR...

Reply:
Yes